hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/fsdataset/impl/NativePmemMappableBlockLoader.java

```diff
@@ -189,4 +190,25 @@ private void verifyChecksumAndMapBlock(POSIX.PmemMappedRegion region,
   public boolean isNativeLoader() {
     return true;
   }
+
+  @Override
+  public MappableBlock getMappableBlock(File file, String bpid)
+      throws IOException {
+    NativeIO.POSIX.PmemMappedRegion region =
+        NativeIO.POSIX.Pmem.mapBlock(file.getAbsolutePath(),
+            file.length(), true);
+    if (region == null)
+      throw new IOException("Failed to restore the block "
+          + file.getName() + " in persistent storage.");
+    String blockId = file.getName();
```

Comment:
Use a dedicated method to parse the file name and return blockId (long type). This method can be inside the super class PmemMappableBlockLoader.